### PR TITLE
Update the Gutenberg release process documentation

### PR DESCRIPTION
**Pull request by ndiego**
## What?
This PR updates the [Gutenberg release documentation](https://developer.wordpress.org/block-editor/contributors/code/release/) to make sure it is up-to-date with current processes and workflows. 

This PR supersedes https://github.com/WordPress/gutenberg/pull/50922. Props to @bph for starting that PR and collaborating on the changes in this PR.  

## Why?
The Gutenberg release docs were out of date and not in line with our current processes. 

Note that this PR only focuses on updating the Gutenberg plugin release information. The rest of the doc is largely unchanged and will likely need further updates in follow-up PRs. Also, this is a living document, and we expect further changes will need to be made in the future. This is merely a start.

### CODE REVIEW

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -1,149 +1,263 @@
...
+* Remove PRs that revert other PRs in the same release if the net change in code is zero. 
+* Remove all PRs that only update the mobile app. The only exception to this rule is if the mobile app pull request also updates functionality for the web.
+* If a subheader only has one PR listed, remove the subheader and move the PR to the next matching subheader with more than one item listed.
 
-There are a couple of ways a release coordinator might be made aware of these bugs:
+#### Creating release candidate patches (cherry-picking)
 
-- Contributors may add the `Backport to Gutenberg RC` label to a closed PR. [Do a search for any of these PRs](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22Backport+to+Gutenberg+RC%22+is%3Aclosed) before publishing the final release.
-- You may receive a direct message or a ping in the #core-editor channel in slack notifying you of PRs that need to be included in the release candidate. Even when this is the case, the `Backport to Gutenberg RC` should be added to the PR.
+QUICK REFERENCE!!!!!!!!
```
Comment (priethor) [inline docs/contributors/code/release.md]: Is this intended? Looks like a placeholder 😅

**Review by priethor — APPROVED** [review]
Fantastic job, both! At last, bringing some clarity to a documentation that currently makes the release look more complex and scary than it actually is. ❤️ 

In that regard, what do you think about adding a `TL;DR;` in a follow-up? I worry folks get discouraged from handling releases by seeing a long documentation and getting the wrong impression. 🤔

File: docs/contributors/code/release.md (hunk 1 above)
Comment (ndiego) [reply docs/contributors/code/release.md]: 🤦‍♂️ Yup, I was thinking about doing "Quick references" in each section that provide the top-level bullet points for those more familiar with the process who just need a reminder of the steps. Precisely what you highlighted 👇

> In that regard, what do you think about adding a TL;DR; in a follow-up? I worry folks get discouraged from handling releases by seeing a long documentation and getting the wrong impression. 🤔

I was thinking about this being in a follow-up, but I might as well get it done in this PR. I'll work on adding that today.

Comment (ndiego) [reply docs/contributors/code/release.md]: This has been fixed, and I added a few "Quick reference" callouts to various sections. We may want to add more in the future, but I think this, plus the addition of the TOC, makes this doc much easier to work through.

Comment (priethor) [reply docs/contributors/code/release.md]: Fantastic! Let's merge!

**Review by priethor — APPROVED** [review]
The updated version with callouts looks even better; let's merge it! 🚀

**Pull request merged**